### PR TITLE
[confighttp] Add option to disable HTTP keep-alives

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
**Description:**

Add option to disable HTTP keep-alives which helps in better load balancing in k8s clusters when otlphttp exports are used to send logs to otlp receivers

**Link to tracking Issue:**  #8260

### CODE REVIEW

**Comment by atoulme** [comment]
Please add a changelog entry.

File: config/confighttp/confighttp.go
Comment (atoulme) [inline config/confighttp/confighttp.go:80]: Why are you using a boolean pointer? Can a boolean do the job?

Comment (kkujawa-sumo) [reply config/confighttp/confighttp.go]: No special reason, boolean is also ok.

**Comment by kkujawa-sumo** [comment]
@bogdandrutu Could you take a look at this pull request?

**Review by astencel-sumo — APPROVED** [review]
This makes sense to me, the change seems pretty straightforward too.

**Review by atoulme — APPROVED** [review]
LGTM. Please rebase off main to fix the conflict.

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
@open-telemetry/collector-maintainers, I believe this is ready to be merged.

File: config/confighttp/confighttp.go
Comment (Aneurysm9) [inline config/confighttp/confighttp.go:79]:
```suggestion
	// for a single HTTP request.
	//
	// WARNING: enabling this option can result in significant overhead establishing a new HTTP(S)
	// connection for every request.  Before enabling this option please consider whether changes
	// to idle connection settings can achieve your goal.
```

Comment (kkujawa-sumo) [reply config/confighttp/confighttp.go:79]: @Aneurysm9 I added this warning

Comment (jpkrohling) [reply config/confighttp/confighttp.go:79]: I wonder if this should be a warning for the user of the collector as well, visible in the collector's logs when this option is enabled.

Comment (codeboten) [reply config/confighttp/confighttp.go:79]: I feel like this warning, if it was visible in the collector logs, would potentially just be noise to someone who's already aware of the configuration they're using and need to have this option disabled.

**Review by codeboten — COMMENTED** [review]
@Aneurysm9 please take a look, the warning has been added.

**Review by codeboten — APPROVED** [review]

**Comment by Aneurysm9** [comment]
> @Aneurysm9 please take a look, the warning has been added.

I would prefer not to add this configuration option at all, but don't feel strongly enough to block this PR, so I'll simply decline to approve it.  

I'm not sure that logging a warning when this option is enabled would be more than noise.  Hopefully someone enabling this option has a better understanding of their environment than we do and understands what they're buying.

**Pull request merged**
